.github/workflows/test.yml

```diff
@@ -40,8 +41,19 @@ jobs:
           - "3.8"
     runs-on: ubuntu-latest
     needs: test
+    timeout-minutes: 12
     name: Test @ Python ${{ matrix.python-version }}
-
+    env:
```

Comment:
@AsifNawaz-cnic why have you changed the installation procedure of python dependencies in that way? With `pip install --upgrade -e .` it is automatically referencing to the setup.py.
Dependabot won't be able to automatically update those package lists in our workflows and this would mean additional manual work.